src/Common/ThreadPool.cpp

```diff
@@ -74,6 +74,8 @@ void ThreadPoolImpl<Thread>::setQueueSize(size_t value)
 {
     std::lock_guard lock(mutex);
     queue_size = value;
+    /// Reserve memory to get rid of allocations
+    jobs.reserve(2 * queue_size);
```

Comment:
Why it is not just `queue_size`?

Reply:
Probably because of heap ...
Need a comment

Reply:
Will be fixed in next commit.